MaterialLib/MPL/Properties/CapillaryPressureSaturation/CapillaryPressureRegularizedVanGenuchten.cpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * Created on April 20, 2020, 9:30 AM
+ */
+
+#include "CapillaryPressureRegularizedVanGenuchten.h"
+
+#include <algorithm>
+#include <cmath>
+
+#include "BaseLib/Error.h"
+#include "MaterialLib/MPL/Medium.h"
+#include "MaterialLib/MPL/VariableType.h"
+#include "MaterialLib/MPL/Utils/CheckVanGenuchtenExponentRange.h"
+
+namespace MaterialPropertyLib
+{
+void checkSaturationRange(const double Sl)
+{
+    if (Sl < 0 || Sl > 1)
+    {
+        OGS_FATAL("The saturation of {:g} is  out of its range of [0, 1]", Sl);
```

Comment:
Error message uses closed interval, but check is for open interval.
Minor: Extra space between "is" and "out".

Reply:
S>1 is not allowed.
Extra space between "is" and "out" was removed.

Reply:
The if condition in line 26 handles unallowed input. The error message corresponding to an input value 1 + 1e-8 would be:
``'The saturation of 1 is out of range of [0, 1].```
Some users may not be able to see the error in it.

Reply:
Changed the output format to {:e}